scipy/spatial/transform/_rotation.pyx

```diff
@@ -1273,6 +1386,186 @@ cdef class Rotation:
+            - array_like with shape (N, W) where each `angle[i]`
+              corresponds to a sequence of Davenport angles describing a 
+              single rotation
+
+        extrinsic : boolean
+            If True, sequence will be extrinsic. If False, sequence will be
+            treated as intrinsic.
+        degrees : bool, optional
+            If True, then the given angles are assumed to be in degrees. 
+            Default is False.
+
+        Returns
+        -------
+        rotation : `Rotation` instance
+            Object containing the rotation represented by the sequence of
+            rotations around given axes with given angles.
+
+        References
+        ----------
+        .. [1] https://en.wikipedia.org/wiki/Euler_angles#Definition_by_intrinsic_rotations
+        .. [2] Shuster, Malcolm & Markley, Landis. (2003). Generalization of
+               the Euler Angles. Journal of the Astronautical Sciences. 51. 123-132. 10.1007/BF03546304.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Davenport angles are a generalization of Euler angles, when we use the
+        canonical basis axes:
+
+        >>> ex = [1, 0, 0]
+        >>> ey = [0, 1, 0]
+        >>> ez = [0, 0, 1]
+
+        Initialize a single rotation with a given axis sequence:
+
+        >>> axes = [ez, ey, ex]
+        >>> r = R.from_davenport(axes, [90, 0, 0], extrinsic=True, degrees=True)
+        >>> r.as_quat().shape
+        (4,)
+
+        It is equivalent to Euler angles in this case:
+
+        >>> r.as_euler('zyx', degrees=True)
+        array([90.,  0., -0.])
+
+        Initialize multiple rotations in one object:
+
+        >>> r = R.from_davenport(axes, [[90, 45, 30], [35, 45, 90]], extrinsic=True, degrees=True)
+        >>> r.as_quat().shape
+        (2, 4)
+
+        Non-canonical axes are possible, and they do not need to be normalized,
+        as long as consecutive axes are orthogonal:
+
+        >>> e1 = [2, 0, 0]
+        >>> e2 = [0, 1, 0]
+        >>> e3 = [1, 0, 1]
+        >>> axes = [e1, e2, e3]
+        >>> r = R.from_davenport(axes, [90, 45, 30], extrinsic=True, degrees=True)
+        >>> r.as_quat().shape
+        (4,)
+
+        """
+        axes = np.asarray(axes)
+        if axes.ndim == 1:
+            axes = axes.reshape([1, 3])
+
+        num_axes = len(axes)
+
+        for axis in axes:
+            if len(axis) != 3:
+                raise ValueError("Axes must be vectors of length 3.")
+
+        if num_axes < 1 or num_axes > 3:
+            raise ValueError("Expected up to 3 axes, got {}".format(num_axes))
+
+        # normalize axes
+        norm = np.repeat(np.linalg.norm(axes, axis=1), 3)
+        axes = axes / norm.reshape(num_axes, 3)
+
+        if num_axes > 1 and abs(np.dot(axes[0], axes[1])) >= 1e-7:
+            raise ValueError("Consecutive axes must be orthogonal.")
+        if num_axes > 2 and abs(np.dot(axes[1], axes[2])) >= 1e-7:
```

Comment:
I made it this way because if there are two axis, then only the first test should be done, while for 3 axes, then both tests should be done.

Do you think there's a better way in this case?

Reply:
It's not better, but if I'm not mistaken, just combining the two conditions with `or` will give exactly the same behavior.

Reply:
I meant exactly same conditions

```
if (num_axes > 1 and abs(np.dot(axes[0], axes[1])) >= 1e-7 or 
    num_axes > 2 and abs(np.dot(axes[1], axes[2])) >= 1e-7):
   ...
```

Reply:
Ah right, changed that!